python/google/protobuf/internal/decoder.py

```diff
@@ -85,18 +85,17 @@
 __author__ = 'kenton@google.com (Kenton Varda)'
 
 import struct
-import sys  ##PY25
-_PY2 = sys.version_info[0] < 3  ##PY25
 from google.protobuf.internal import encoder
 from google.protobuf.internal import wire_format
+from google.protobuf.internal.utils import bytestr_to_string, b
 from google.protobuf import message
 
 
 # This will overflow and thus become IEEE-754 "infinity".  We would use
 # "float('inf')" but it doesn't work on Windows pre-Python-2.6.
-_POS_INF = 1e10000
-_NEG_INF = -_POS_INF
-_NAN = _POS_INF * 0
+_POS_INF = float('inf')
+_NEG_INF = float('-inf')
+_NAN = float('nan')
```

Comment:
Maybe it would be better to split python bindings into different projects: one for python 2.x one for 3.x and one for common parts?

It's ridiculous how much time was spent on such simple issue :) New features are not introduced that often to consider supporting separate projects an impossible task.